account_invoice_alternate_payer/models/account_move.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2018 Eficent Business and IT Consulting Services, S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from contextlib import contextmanager
+
+from odoo import _, api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    alternate_payer_id = fields.Many2one(
+        "res.partner",
+        string="Alternate Payer",
+        help="If set, this will be the partner that we expect to pay or to "
+        "be paid by. If not set, the payor is by default the "
+        "commercial",
+    )
+
+    # pylint: disable=W8110
+    @api.depends("alternate_payer_id")
+    def _compute_bank_partner_id(self):
+        super()._compute_bank_partner_id()
+        for move in self.filtered(lambda r: r.alternate_payer_id and r.is_outbound()):
+            move.bank_partner_id = move.alternate_payer_id
+
+    @api.onchange("alternate_payer_id")
+    def _onchange_alternate_payer_id(self):
+        self._onchange_partner_id()
```

Comment:
```suggestion
        return self._onchange_partner_id()
```
it may return a warning message somewhere, now or in future.